hibernate-core/src/main/java/org/hibernate/action/internal/AbstractEntityInsertAction.java

```diff
@@ -132,6 +138,27 @@ public final void makeEntityManaged() {
 				getPersister(),
 				isVersionIncrementDisabled
 		);
+		if ( isEarlyInsert() ) {
+			final SharedSessionContractImplementor session = getSession();
+			final PersistenceContext persistenceContext = session.getPersistenceContextInternal();
+			Object[] objects = getState();
+			for ( int i = 0; i < objects.length; i++ ) {
+				if ( objects[i] instanceof PersistentCollection<?> ) {
```

Comment:
It is uncommon to specify generic type in `instanceof` expression for generic type has been erased during runtime, but Java compiler would not bother warning about this for it is purely unharmful.